src/ia/js/IAPage.js

```diff
@@ -167,25 +167,58 @@
                         $(".special-permissions__toggle-view").hide();
                     });
 
+                    // Screenshots TODO:
+                    // - Notify the user of any errors.
+                    // - Tell the user if the service is down.
+                    
+                    // Check if the screenshot exists or not.
                     $(".ia-single--image-container img").error(function() {
                         if (ia_data.live.dev_milestone !== "live") {
-                            $(".ia-single--screenshots").addClass("hide");
-                            $(".ia-single--left").removeClass("ia-single--left").addClass("ia-single--left--wide");
-                            $(".dev_milestone-container__body").removeClass("hide");
-
-                            // Set the panels height to the tallest one's height
-                            var max_height = 0;
-                            $(".dev_milestone-container").each(function(idx) {
-                                if ($(this).height() > max_height) {
-                                    max_height = $(this).height();
-                                }
-                            });
-
-                            $(".dev_milestone-container").height(max_height);
+                            // Hide the faux-window if the screenshot doesn't exist.
+                            $(".ia-single--screenshots").hide();
+                            $(".ia-single--right").addClass("dashed-border");
 
                             page.imgHide = true;
                         }
                     });
+
+                    // Saves the screenshot to S3.
+                    $(".save-screenshot--button").click(function() {
+                        $.post("/screenshot/save/" + ia_data.live.id, function(data) {
+                            // Hide the save button.
+                            $(".save-screenshot--button").addClass("hide");
+                            // Put in the image from S3.
+                            $(".ia-single--image-container img").attr("src", "https://images.duckduckgo.com/iu/?u=" + encodeURIComponent(data.screenshots.index));
+                        });
+                    });
+
+                    // Generate a screenshot when the button is clicked.
+                    $(".generate-screenshot--button").click(function() {
```

Comment:
Not exactly the same - the way you've written it the event is bound to the element, and when the page is loaded that element doesn't exist yet, it's created afterwards after adding the screens handlebars template. So JQuery might not see it.

The way I suggested you to write binds the event to the body, so when you click it will always trigger.